lib/ui/painting/image_filter.cc

```diff
@@ -40,6 +40,29 @@ fml::RefPtr<ImageFilter> ImageFilter::Create() {
   return fml::MakeRefCounted<ImageFilter>();
 }
 
+static const std::array<SkSamplingOptions, 4> filter_qualities = {
+    SkSamplingOptions(SkFilterMode::kNearest, SkMipmapMode::kNone),
+    SkSamplingOptions(SkFilterMode::kLinear, SkMipmapMode::kNone),
+    SkSamplingOptions(SkFilterMode::kLinear, SkMipmapMode::kLinear),
+    SkSamplingOptions(SkCubicResampler{1 / 3.0f, 1 / 3.0f}),
+};
+
+SkSamplingOptions ImageFilter::SamplingFromIndex(int filterQualityIndex) {
+  if (filterQualityIndex < 0)
+    return filter_qualities.front();
+  else if (((size_t)filterQualityIndex) >= filter_qualities.size())
+    return filter_qualities.back();
+  else
+    return filter_qualities[filterQualityIndex];
```

Comment:
I believe the style guide requires curly braces here: https://google.github.io/styleguide/cppguide.html#Conditionals
```suggestion
  if (filterQualityIndex < 0) {
    return filter_qualities.front();
  } else if (((size_t)filterQualityIndex) >= filter_qualities.size()) {
    return filter_qualities.back();
  } else {
    return filter_qualities[filterQualityIndex];
  }
```